channeldb/payments.go

```diff
@@ -624,6 +633,24 @@ func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
 			return err
 		}
 
+		// Counting the total number of payments is expensive, since we
+		// literally have to traverse the cursor linearly, which can
+		// take quite a while. So it's an optional query parameter.
+		if query.CountTotal {
+			var totalPayments uint64
+			err := indexes.ForEach(func(_, _ []byte) error {
```

Comment:
nit: we could attempt to `ForAll` here if supported (if `indexes` is a `kvdb.ExtendedRBucket`) and fall back to using `ForEach` otherwise.

Reply:
Great idea, added!